### ISSUE TITLE
🐛 BUG: site config doesn't seem to be respected in bundled assets

### ISSUE THREAD

**Issue by jeffjose** [snowpackjs/astro#963]
### What package manager are you using?

yarn

### What operating system are you using?

Linux

### Describe the Bug

Allow astro site to be deployable to a subfolder, via a configuration in `astro.config.mjs`.

The final state is an astro app `app` deployed at `https://example.com/app`, with `/app` prefixed to the resource paths.

### Steps to Reproduce

1. `npm init astro` using template svelte
2. Update hypothetical `buildOptions.pathPrefix = app` in `astro.config.mjs`
2. `yarn build`
3. `cat dist/index.html | grep _astro`

**EXPECTED**
```html
<!DOCTYPE html><html lang="en"><head><link rel="stylesheet" href="/app/_astro/index-2hTXm5.css"
```

**ACTUAL**
```html
<!DOCTYPE html><html lang="en"><head><link rel="stylesheet" href="_astro/index-2hTXm5.css"
```

**Comment by jasikpark** [comment]
One idea I had for config would be
```ts
{
  site: string | { base : string, path : string }
 }
```
so you'd either specify
```ts
{
  site: "example.com/newapp"
}
```
or
```ts
{
  site: { 
    path: "/newapp" 
  }
}
```

**Comment by jeffjose** [comment]
I like the @jasikpark's proposal.

**Comment by jeffjose** [comment]
Gentle ping :)

**Comment by drwpow** [comment]
Because of its relation to #1073, we should prioritize a solution for this. Hopefully in the next RFC call!

**Comment by matthewp** [comment]
I'd love to discuss this on the RFC call tomorrow: https://github.com/snowpackjs/astro/issues/1144

**Comment by matthewp** [comment]
RFC Discussion:

* `Astro.site` should probably be doing this already, so probably a bug.
* Astro.site might be overloaded. We should revisit it and possibly split into multiple config properties.
  * There's a need for a base path
  * There's a need for Astro.site for a site map?
* Work on this RFC for a proposal to solve the dual need.

**Comment by FredKSchott** [comment]
If this is an RFC, it needs to use something more along the lines of the RFC template. I'm going to rename this as a bug and then if anyone wants to submit an RFC to discuss, we can do that there.

**Comment by FredKSchott** [comment]
Also @matthewp this is a good reminder to confirm that your hoisted scripts bundling honors this in the script src output!

**Comment by matthewp** [comment]
We are tracking fixing subfolder hosting either alongside 0.21 or very shortly thereafter. This use-case is very important.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
